_posts/2021-01-13-hapi-fhir.md

```diff
@@ -0,0 +1,454 @@
+        return registration;
+    }
+}
+```
+
+Once your server has started, 
+Open up your Postman and enter this URL http://localhost:8080/metadata. 
+A capability statement will appear just like the screenshot below. 
+    Congratulations you are now FHIR enabled!
+
+<img class="image fit" alt="postman" src="/img/2021-01-23-hapi-fhir/postman.png">
+
+
+### Resources
+
+The Patient provider is where the FHIR Patient behaviour is configured. 
+HAPI uses annotations to indicate what service the procedures provide. 
+The procedure createPatient in the diagram is annotated with @Create which indicates it handles POST/create. 
+This procedure then uses a PatientDAO (DAO - data access object) class which uses Spring Data to persist the Patient resource in the MongoDB.
+
+  
+
+```Java
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.annotation.*;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.StringParam;
+import ca.uhn.fhir.rest.param.TokenParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import com.example.fhirexample.dao.patient.IPatient;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class PatientProvidor implements IResourceProvider {
+
+    @Autowired
+    private FhirContext ctx;
+
+    @Autowired
+    private IPatient patientDao;
+
+    private Map<String, Patient> patientMap = new HashMap<String, Patient>();
+
+    public  PatientProvidor() {
+    
+    }   
+    
+    @Override
+    public Class<? extends IBaseResource> getResourceType() {
+        return  Patient.class;
+    }
+
+    @Search
+    public List<Resource> searchPatient(HttpServletRequest request,
+                                        @OptionalParam(name= Patient.SP_BIRTHDATE) DateRangeParam birthDate,
+                                        @OptionalParam(name = Patient.SP_FAMILY) StringParam familyName,
+                                        @OptionalParam(name= Patient.SP_GENDER) StringParam gender,   
+                                        @OptionalParam(name= Patient.SP_GIVEN) StringParam givenName,
+                                        @OptionalParam(name = Patient.SP_IDENTIFIER) TokenParam identifier,
+                                        @OptionalParam(name= Patient.SP_NAME) StringParam name,
+                                        @OptionalParam(name = Patient.SP_RES_ID) TokenParam resid) {
+
+        List<Resource> results = patientDao.search(ctx, birthDate, familyName, gender, givenName, identifier, name);
```

Comment:
No need to create the "results" variable here:
"return patientDao.search(ctx, birthDate, familyName, gender, givenName, identifier, name);"

Same applies for all methods in this class.